### PR TITLE
Add basic CLI

### PR DESCRIPTION
**Pull request by sd109**
Currently supports host and port args (which can also be set by `S3_ACTIVE_STORAGE_{HOST,PORT}` env vars. Is it worth adding some kind of `--debug` flag for setting log level instead of falling back to `RUST_LOG`?

### CODE REVIEW

**Comment by sd109** [comment]
Example of current cli output:
```
S3 Active Storage Proxy command line interface

Usage: s3-active-storage [OPTIONS]

Options:
      --host <HOST>  [env: S3_ACTIVE_STORAGE_HOST=] [default: 0.0.0.0]
      --port <PORT>  [env: S3_ACTIVE_STORAGE_PORT=] [default: 8080]
  -h, --help         Print help
```

**Comment by sd109** [comment]
Addresses #14

File: src/main.rs [hunk 1]
```diff
@@ -34,15 +35,26 @@ mod operations;
 mod s3_client;
 mod validated_json;
 
+/// S3 Active Storage Proxy command line interface
+#[derive(Debug, Parser)]
+struct CommandLineArgs {
+    #[arg(long, default_value = "0.0.0.0", env = "S3_ACTIVE_STORAGE_HOST")]
+    host: String,
+    #[arg(long, default_value = "8080", env = "S3_ACTIVE_STORAGE_PORT")]
+    port: String,
```
Comment (markgoddard) [inline src/main.rs:44]: Let's make it a u16 to get some better input validation.

```suggestion
    #[arg(long, default_value = 8080, env = "S3_ACTIVE_STORAGE_PORT")]
    port: u16,
```

Comment (markgoddard) [reply src/main.rs:44]: Or there's this: https://github.com/clap-rs/clap-port-flag

File: src/main.rs [hunk 2]
```diff
@@ -34,15 +35,26 @@ mod operations;
 mod s3_client;
 mod validated_json;
 
+/// S3 Active Storage Proxy command line interface
+#[derive(Debug, Parser)]
+struct CommandLineArgs {
+    #[arg(long, default_value = "0.0.0.0", env = "S3_ACTIVE_STORAGE_HOST")]
+    host: String,
```
Comment (markgoddard) [inline src/main.rs:42]: How about some short help strings?

**Review by markgoddard — CHANGES_REQUESTED** [review]

**Comment by sd109** [comment]
Newer CLI help output with added doc strings:
```
S3 Active Storage Proxy command line interface

Usage: s3-active-storage [OPTIONS]

Options:
      --host <HOST>  The IP address on which the proxy should listen [env: S3_ACTIVE_STORAGE_HOST=] [default: 0.0.0.0]
      --port <PORT>  The port to which the proxy should bind [env: S3_ACTIVE_STORAGE_PORT=] [default: 8080]
  -h, --help         Print help
```

**Review by markgoddard — APPROVED** [review]

**Pull request merged**